### PR TITLE
Make it easier to evaluate Bumblebee

### PR DESCRIPTION
**Pull request by yoctozepto**
Add some more docs and adapt the code to lower the entry barrier.

### CODE REVIEW

**Comment by andybotting** [comment]
This looks really good at first glance. I'll take a closer look at work next week and merge

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,7 @@ BUILDER_ARGS=
 build:
 	echo "Derived image tag: $(DESCRIBE)"
 	echo "Actual image tag: $(IMAGE_TAG)"
-	$(BUILDER) build -f docker/Dockerfile $(BUILDER_ARGS) -t $(IMAGE) .
+	$(BUILDER) build $(BUILDER_ARGS) -t $(IMAGE) .
```
Comment (bahnwaerter) [inline Makefile:14]: Instead of creating a symbolic link for the Docker file, you could specify more build information in the Docker file's service description:
```yaml
build:
  context: .
  dockerfile: docker/Dockerfile
```
Such a change would consider the actual directory structure. Then you could revert the change in this Makefile.

File: docker-compose.yml [hunk 2]
```diff
@@ -53,34 +63,35 @@ services:
...
-    - redis:redis
     volumes:
     - .:/app
 
   mariadb:
     image: mariadb
-    env_file:
-    - .env
+    environment:
+      MARIADB_ROOT_PASSWORD: toor
```
Comment (bahnwaerter) [inline docker-compose.yml:80]: In addition, we could specify default inline values for service-dependent environment variables, as well as our service-independent global variables like `DB_PASSWORD`:
```suggestion
      MARIADB_ROOT_PASSWORD: ${DB_PASSWORD:-toor}
```
This suggestion would simplify the understanding for developers who are not familiar with Docker's [environment variables precedence](https://docs.docker.com/compose/envvars-precedence/). Further, this suggestion automatically documents environment variables that could be set in a Docker's `*.env` file if the inlined default values shouldn't be evaluated.

File: initdb/01.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+CREATE DATABASE `bumblebee`;
```
Comment (bahnwaerter) [inline initdb/01.sql:1]: For safety reasons, you could add a check to avoid modifications if the table already exists:
```suggestion
CREATE DATABASE IF NOT EXISTS `bumblebee`;
```

File: initdb/01.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
...
+CREATE DATABASE `keycloak`;
```
Comment (bahnwaerter) [inline initdb/01.sql:2]: For safety reasons, you could add a check to avoid modifications if the table already exists:
```suggestion
CREATE DATABASE IF NOT EXISTS `keycloak`;
```

**Review by bahnwaerter — CHANGES_REQUESTED** [review]
Nice work. Finally there are changes that allow an automatic (failure-free) setup of a Bumblebee environment from scratch.

I reviewed your changes and found some lines where alternative solutions are possible. Feel free to think about it and choose one solution or the other.

@andybotting: Do you have something to add?

**Comment by andybotting** [comment]
@bahnwaerter Thanks. These all sound like good suggestions. We're managing the code for this on our gerrit, so I'll look at including your changes through that.

**Comment by andybotting** [comment]
I've pushed some changes that build on this work, so I'll close this PR now. Thanks @yoctozepto and @bahnwaerter!
